### PR TITLE
auth: PKCS11 improvements

### PR DESCRIPTION
**Pull request by Habbie**
### Short description
* ECDSA support
* tests upgraded to softhsm2
* tests enabled in CircleCI

TODO:
* [x] code could use a bit of cleanup now (centralise the two Curve parameter strings; make sure we don't crash on unmapped algorithms; ...)
* [x] documentation is wildly outdated and contains a confusing deprecation message about 4.0.0
* [ ] ~add per algo tests?~ maybe later

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: pdns/pdnsutil.cc [hunk 1]
```diff
@@ -1682,7 +1682,7 @@ static bool showZone(DNSSECKeeper& dk, const DNSName& zone, bool exportDS = fals
         cout<<"ID = "<<value.second.id<<" ("<<DNSSECKeeper::keyTypeToString(value.second.keyType)<<")";
       }
       if (value.first.getKey()->getBits() < 1) {
-        cerr<<" <key missing or defunct>" <<endl;
+        cout<<" <key missing or defunct>" <<endl;
```
Comment (rgacogne) [inline pdns/pdnsutil.cc:1685]: I wonder if the move from `stderr` to `stdout` really makes sense?

Comment (Habbie) [reply pdns/pdnsutil.cc:1685]: During regression tests, the output of pdnsutil is passed through `grep` and friends to get key IDs, and this message prints to the terminal repeatedly with no context. Changing to `cout` avoids that. In general I'm not sure it ever makes sense to mix stdout and stderr in the same output line.

Does that help?

Comment (rgacogne) [reply pdns/pdnsutil.cc:1685]: Yes and no! I'm now wondering if there are cases where this message is actually helpful? :)

Comment (Habbie) [reply pdns/pdnsutil.cc:1685]: Perhaps if we add `run pdnsutil hsm create-key?` to it, I'll do that :)

Comment (Habbie) [reply pdns/pdnsutil.cc]: pushed that

**Pull request merged**